src/com/intellij/stats/ngram/NGramContributor.kt

```diff
@@ -0,0 +1,54 @@
+package com.intellij.stats.ngram
+
+import com.intellij.codeInsight.completion.CompletionContributor
+import com.intellij.codeInsight.completion.CompletionParameters
+import com.intellij.codeInsight.completion.CompletionResult
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.codeInsight.lookup.LookupElementWeigher
+import com.intellij.openapi.project.Project
+import com.intellij.psi.search.GlobalSearchScope
+
+class NGramContributor : CompletionContributor() {
+    override fun fillCompletionVariants(parameters: CompletionParameters, result: CompletionResultSet) {
+        result.runRemainingContributors(parameters, { completionResult ->
+            val wrapped = CompletionResult.wrap(completionResult.lookupElement,
+                    completionResult.prefixMatcher,
+                    completionResult.sorter.weigh(NaiveNGramWeigher(parameters)))
```

Comment:
We have to create a weigher before the `runRemainingContributors` call instead of creating a new weigher for each lookup element. This is my error. Please, extract the constructor call.